src/components/LocationSearch/LocationSearch.test.jsx

```diff
@@ -0,0 +1,56 @@
+import { describe, it, expect, vi } from "vitest";
+import { render, screen, fireEvent } from "@testing-library/react";
+import LocationSearch from "./LocationSearch";
+
+describe("</ LocationSearch", () => {
+  const renderLocationSearch = (props = {}) => {
+    const defaultProps = {
+      error: "",
+      locations: ["New York", "São Paulo"],
+      isLocationSearchOpen: true,
+      handleLocationSearch: vi.fn(),
+      handleLocationSearchVisibility: vi.fn(),
+    };
+    render(<LocationSearch {...defaultProps} {...props} />);
+  };
+
+  it("renders the error message", () => {
+    renderLocationSearch({ error: "some error here" });
+  });
+
+  it("renders LocationSearch correctly", () => {
+    renderLocationSearch();
+
+    //Checks if the input field is render
+    expect(screen.getByPlaceholderText("search location")).toBeInTheDocument();
+
+    //Checks if the serach button is render
+    expect(screen.getByText("Search")).toBeInTheDocument();
+
+    //Checks if the location list is render
+    expect(screen.getByText("New York")).toBeInTheDocument();
+    expect(screen.getByText("São Paulo")).toBeInTheDocument();
+  });
+
+  it("renders location search form based on isLocationSearchOpen prop", () => {
+    renderLocationSearch({ isLocationSearchOpen: true });
+
+    expect(screen.getByPlaceholderText("search location")).toBeVisible();
+  });
+
+  it("calls handleLocationSearch when submitting a valid search term", () => {
+    const handleLocationSearch = vi.fn();
+    renderLocationSearch({ handleLocationSearch: handleLocationSearch });
```

Comment:
You can use shorthand here, always it's the same variable name for the key object, you can omit:

```suggestion
    renderLocationSearch({ handleLocationSearch });
```